src/test/java/com/google/devtools/build/lib/skylark/SkylarkIntegrationTest.java

```diff
@@ -787,7 +787,8 @@ public void testRuleClassImplicitOutputs() throws Exception {
     scratch.file(
         "test/skylark/extension.bzl",
         "def custom_rule_impl(ctx):",
-        "  files = [ctx.outputs.lbl, ctx.outputs.list, ctx.outputs.str]",
+        "  files = [ctx.outputs.lbl, ctx.outputs.str]",
+        "  files += ctx.outputs.list",
```

Comment:
Generally, I'm all for attempting to prevent breakages, but I want to lay out the following scenario, as it would occur without this change:

Rule takes a list attribute
Output function/dictionary specifies a template substitution for an output based on the list attribute

In two invocation cases:
Rule is invoked with an empty list of values
Bazel will error with the following when the output is accessed by name:
```
No attribute '<list output attribute name>' in outputs. Make sure you declared a rule output with this name.
```

Or

Rule is invoked with a plural list of values (more than one entry in a list)
Bazel will error with the following:
```
java.lang.RuntimeException: Unrecoverable error while evaluating node 'PACKAGE:' (requested by nodes 'TARGET_PATTERN::empty')
        at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:438)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:501)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: expected one element but was: <a.out, b.out>
```

Yes, the fix breaks anyone who has specified a list attribute in a skylark rule that has supplied a singleton list and expected it to be resolved to a singular object by forcing them to either correctly specify their attribute as a singular type or iterate over the resulting list value, but isn't this more correct?

I've attached a small workspace demonstrating both of the failures in packages `empty` and `plural` with a small custom rule, buildable with `bazel build empty` and `bazel build plural`
[confer.tar.gz](https://github.com/bazelbuild/bazel/files/970927/confer.tar.gz)